File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, height: List[int]) -> int:
        l, r = 0, len(height)-1

        max_left = height[l]
        max_right = height[r]

        trapped_water = 0

        while l < r:
            if max_left <= max_right:
                l += 1
                max_left = max(max_left, height[l])
                trapped_water += max_left - height[l]
            elif max_left > max_right:
                r -= 1
                max_right = max(max_right, height[r])
                trapped_water += max_right - height[r]
        
        return trapped_water
        

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        //left pointer, right pointer = left pointer + 1
        //volume = smallerColumn height * (r - l)
        //if volume is greater than max volume
        int volume = 0;
        int max = 0;
        for(int i = 0; i < heights.length -1; i++){
            for(int j = i + 1; j < heights.length; j++){
                if(heights[i] < heights[j]){
                    volume = heights[i] * (j - i);
                }else if(heights[i] >= heights[j]){
                    volume = heights[j] * (j - i);
                }
                if(volume > max){
                    max = volume;
                }
            }
        }
        return max;
    }
}
